fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -775,6 +775,34 @@ public synchronized List<CommitCheckAsync> getCommitChecks(@Nonnull Predicate<Co
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Remove commit checks based on a predicate, once the commit check completes.
+     * <p>
+     *     This waits until the commit check completes before removing to ensure that, if it is doing something you can
+     *     ensure that its activity completes before it is removed.
+     * </p>
+     * @param filter if this returns {@code true} the commit check will be removed when it completes
+     * (successfully or unsuccessfully)
+     * @param shouldSwallow a predicate to swallow exceptions, if desired
+     * (see {@link MoreAsyncUtil#swallowException(CompletableFuture, Predicate)})
+     * @return a list of futures that will complete when the associated {@link CommitCheck} has completed, and been
+     * removed from the list of commit checks.
+     */
+    @API(API.Status.INTERNAL)
+    public synchronized List<CompletableFuture<Void>> removeCommitChecks(
+            @Nonnull Function<CommitCheckAsync, Boolean> filter,
+            @Nonnull Predicate<Throwable> shouldSwallow) {
+        return commitChecks.entrySet().stream()
+                .filter(entry -> filter.apply(entry.getValue()))
+                .map(entry -> MoreAsyncUtil.swallowException(entry.getValue().checkAsync(), shouldSwallow)
+                        .whenComplete((result, err) -> removeCommitCheck(entry.getKey())))
+                .collect(Collectors.toList());
+    }
+
+    private synchronized void removeCommitCheck(String key) {
+        commitChecks.remove(key);
```

Comment:
I'd be a little concerned about a `ConcurrentModificationException`s here, if `removeCommitCheck` attempts to remove an element while the `stream` is still going over it. The alternative is probably to do something like (1) collect the filtered commit checks into a list and then (2) remove them only once the iteration is done.

Reply:
`removeCommitCheck` and `removeCommitChecks` are both `synchronized`, so it won't be able to remove the check while it is being iterated over, unless I'm missing something